Observer/NonBoltOrderObserver.php

```diff
@@ -0,0 +1,229 @@
+                return;
+            }
+
+            $quote = $this->quoteRepository->get($order->getQuoteId());
+            if (!$quote) {
+                $this->metricsClient->processCountMetric("non_bolt_order_creation.no_quote", 1);
+                return;
+            }
+
+            $items = $quote->getAllVisibleItems();
+            if (!$items) {
+                $this->metricsClient->processCountMetric("non_bolt_order_creation.no_items", 1);
+                return;
+            }
+
+            $this->handleMissingName($quote);
+
+            $cart = $this->cartHelper->buildCartFromQuote($quote, $quote, $items, null, true, false);
+            $customer = $quote->getCustomer();
+            $storeId = $order->getStoreId();
+            $result = $this->createNonBoltOrder($cart, $customer, $storeId);
+            if ($result != 200) {
+                $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
+                return;
+            }
+        } catch (Exception $exception) {
+            $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
+            $this->bugsnag->notifyException($exception);
+            return;
+        } catch (Error $error) {
+            // catch errors so failures here don't prevent orders from being created
+            $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
+            $this->bugsnag->notifyException($error);
+            return;
+        }
+
+        $this->metricsClient->processCountMetric("non_bolt_order_creation.success", 1);
+    }
+
+    /**
+     * Call Bolt Create Non-Bolt Order API
+     *
+     * @param array $cart
+     * @param CustomerInterface $customer
+     * @param int $storeId
+     * @return Response|int
+     * @throws LocalizedException
+     * @throws Zend_Http_Client_Exception
+     */
+    protected function createNonBoltOrder($cart, $customer, $storeId)
+    {
+        $apiKey = $this->configHelper->getApiKey($storeId);
+
+        $phone = $this->getPhone($cart);
+        $requestData = $this->dataObjectFactory->create();
+        $requestData->setApiData([
+            'cart' => $cart,
+            'user_identifier' => [
+                'email' => $customer->getEmail(),
+                'phone' => $phone,
+            ],
+            'user_identity' => [
+                'first_name' => $customer->getFirstname(),
+                'last_name' => $customer->getLastname(),
+            ],
+        ]);
+        $requestData->setDynamicApiUrl(ApiHelper::API_CREATE_NON_BOLT_ORDER);
+        $requestData->setApiKey($apiKey);
+        $requestData->setStatusOnly(true);
+
+        $request = $this->apiHelper->buildRequest($requestData);
+        return $this->apiHelper->sendRequest($request);
+    }
+
+    /**
+     * Get phone number from the cart if it exists
+     *
+     * @param array $cart
+     * @return string
+     */
+    protected function getPhone($cart)
```

Comment:
I believe this whole function could be replaced with `return $cart['shipments'][0]['shipping_address']['phone'] ?? null;` In fact, since you're returning null anyways for it not being set you might be able to skip the coalescing altogether.

That's a php7 thing, so it may break on older instances, in which case `isset()` might be the ticket.

Reply:
I originally had that, but got an error when the key was missing

Reply:
interesting. I wonder if it's the `[0]` that breaks it

Reply:
Added a test-- it's because if the `'shipments'` key isn't even set, then it will break. I.e. it will work if `$cart['shipments']==null`, but not if `$cart` doesn't have the key `shipments`